src/main/java/edu/ucsb/courses/controllers/ScheduleItemController.java

```diff
@@ -0,0 +1,141 @@
+package edu.ucsb.courses.controllers;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import edu.ucsb.courses.documents.Course;
+import edu.ucsb.courses.documents.Section;
+import edu.ucsb.courses.documents.TimeLocation;
+import edu.ucsb.courses.entities.Schedule;
+import edu.ucsb.courses.entities.ScheduleItem;
+import edu.ucsb.courses.repositories.ScheduleItemRepository;
+import edu.ucsb.courses.repositories.ArchivedCourseRepository;
+import edu.ucsb.courses.repositories.ScheduleRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+
+@RestController
+@RequestMapping("/api/public")
```

Comment:
Should this be under `/api/public`?

Reply:
It should not.  It should be under /api/member.  I've created a separate issue for them to fix this. in the ScheduleController.   For this controller, however, I'm going to hold the line and say:  FIX IT NOW.